frontend/src/metabase-lib/display.ts

```diff
@@ -0,0 +1,160 @@
+import * as Lib from "metabase-lib";
+import type {
+  CardDisplayType,
+  VisualizationSettings,
+} from "metabase-types/api";
+import { TYPE } from "metabase-lib/types/constants";
+import { isa } from "metabase-lib/types/utils/isa";
+
+type DefaultDisplay = {
+  display: CardDisplayType;
+  settings?: Partial<VisualizationSettings>;
+};
+
+export const defaultDisplay = (query: Lib.Query): DefaultDisplay => {
+  const { isNative } = Lib.queryDisplayInfo(query);
+
+  if (isNative) {
+    return { display: "table" };
+  }
+
+  const stageIndex = -1;
+  const aggregations = Lib.aggregations(query, stageIndex);
+  const breakouts = Lib.breakouts(query, stageIndex);
+
+  if (aggregations.length === 0 && breakouts.length === 0) {
+    return { display: "table" };
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 0) {
+    return { display: "scalar" };
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 1) {
+    const [{ columnInfo }] = getBreakoutColumnInfos(
+      breakouts,
+      query,
+      stageIndex,
+    );
+
+    if (isState(columnInfo)) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "region",
+          "map.region": "us_states",
+        },
+      };
+    }
+
+    if (isCountry(columnInfo)) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "region",
+          "map.region": "world_countries",
+        },
+      };
+    }
+  }
+
+  if (aggregations.length >= 1 && breakouts.length === 1) {
+    const [{ breakout, columnInfo }] = getBreakoutColumnInfos(
+      breakouts,
+      query,
+      stageIndex,
+    );
+
+    if (isDate(columnInfo)) {
+      const info = Lib.displayInfo(query, stageIndex, breakout);
+
+      if (info.isTemporalExtraction) {
+        return { display: "bar" };
+      }
+
+      return { display: "line" };
+    }
+
+    const binning = Lib.binning(breakout);
+    const isBinned = binning !== null;
+
+    if (isBinned) {
+      return { display: "bar" };
+    }
+
+    if (isCategory(columnInfo)) {
+      return { display: "bar" };
+    }
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 2) {
+    const infos = getBreakoutColumnInfos(breakouts, query, stageIndex);
+
+    const isAnyBreakoutDate = infos.some(({ columnInfo }) => {
+      return isDate(columnInfo);
+    });
+    if (isAnyBreakoutDate) {
+      return { display: "line" };
+    }
+
+    const areBreakoutsCoordinates = infos.every(({ columnInfo }) => {
+      return isCoordinate(columnInfo);
+    });
+    if (areBreakoutsCoordinates) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "grid",
+        },
+      };
+    }
+
+    const areBreakoutsCategories = infos.every(({ columnInfo }) => {
+      return isCategory(columnInfo);
+    });
+    if (areBreakoutsCategories) {
+      return { display: "bar" };
+    }
+  }
+
+  return { display: "table" };
+};
+
+const getBreakoutColumnInfos = (
```

Comment:
Nit: I don't think "information" has a plural. I understand what you wanted to convey, but it just reads wrong. I'd go with `getBreakoutColumnInfo` or `getBreakoutAndColumnInfo` if you want to be really specific.

Reply:
Renamed to `getBreakoutsWithColumnInfo`: 4d78f22d9a5cf7973b4c5b74fd01c8f508b30abb